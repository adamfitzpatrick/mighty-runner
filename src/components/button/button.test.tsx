import { h } from 'preact'
import deep from 'preact-render-spy'
import Button from '.'

describe('Button component', () => {
  let onClick: jest.Mock

  beforeEach(() => {
    onClick = jest.fn()
  })

  function buildButton (ariaLabel?: string, primary?: boolean, icon?: boolean) {
    return deep(
      <Button
        onClick={onClick}
        ariaLabel={ariaLabel}
        primary={primary}
        icon={icon}
      >
        Stuff
      </Button>,
      { depth: 2 }
    )
  }

  test('should render correctly', () => {
    const button = buildButton()
    expect(button).toMatchSnapshot()
  })

  test('should call the click handler on a click', () => {
    const button = buildButton()
    button.simulate('click')
    expect(onClick).toHaveBeenCalled()
  })

  test('should render with an aria label', () => {
    const button = buildButton('label')
    expect(button).toMatchSnapshot()
  })

  test('should render as primary button', () => {
    const button = buildButton(undefined, true)
    expect(button).toMatchSnapshot()
  })

  test('should render as an icon button', () => {
    const button = buildButton(undefined, undefined, true)
    expect(button).toMatchSnapshot()
  })
})
