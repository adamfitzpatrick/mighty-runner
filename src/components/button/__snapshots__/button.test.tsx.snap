// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render as an icon button 1`] = `
preact-render-spy (1 nodes)
-------
<button
  class="button icon"
  onClick={[Function mockConstructor]}
>
  <span class="buttonFocusIndicator"></span>
  Stuff
</button>

`;

exports[`Button component should render as primary button 1`] = `
preact-render-spy (1 nodes)
-------
<button
  class="button primary"
  onClick={[Function mockConstructor]}
>
  <span class="buttonFocusIndicator"></span>
  Stuff
</button>

`;

exports[`Button component should render correctly 1`] = `
preact-render-spy (1 nodes)
-------
<button
  class="button"
  onClick={[Function mockConstructor]}
>
  <span class="buttonFocusIndicator"></span>
  Stuff
</button>

`;

exports[`Button component should render with an aria label 1`] = `
preact-render-spy (1 nodes)
-------
<button
  class="button"
  onClick={[Function mockConstructor]}
  aria-label="label"
>
  <span class="buttonFocusIndicator"></span>
  Stuff
</button>

`;
